analyze/aps-cycle.c

```diff
@@ -780,7 +780,17 @@ static bool applicable_for_circularity_check(INSTANCE *instance)
     switch (Declaration_KEY(node))
     {
     case KEYformal:
-      return false;
+    {
+      switch (ABSTRACT_APS_tnode_phylum(tnode_parent(node)))
+      {
+      case KEYPattern:
+        // formals used inside match patterns are allowed to be
+        // involved in a cycle.
+        return false;
+      default:
+        break;
+      }
+    }
```

Comment:
I think this change fixes a bug.  It would clearer to have a test/example that this change allows to work.